### ISSUE TITLE
[data grid] Row grouping does not renders child rows in a group when sortingMode is server

### ISSUE THREAD

**Issue by eXist-FraGGer** [mui/mui-x#5650]
### Order ID 💳

11111

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

So the issue is that the rows are rendered at the top and groups at the bottom when the sortingMode is server

You can see an example (Try to comment sortingMode):

https://codesandbox.io/s/data-grid-premium-forked-oyuuwx?file=/src/App.tsx

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

Google Chrome
Version 103.0.5060.134 (Official Build) (arm64)

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
System:
    OS: macOS 12.5
  Binaries:
    Node: 14.19.0 - ~/.nvm/versions/node/v14.19.0/bin/node
    Yarn: 1.22.17 - /opt/homebrew/bin/yarn
    npm: 8.6.0 - ~/.nvm/versions/node/v14.19.0/bin/npm
  Browsers:
    Chrome: 103.0.5060.134
    Edge: Not Found
    Firefox: Not Found
    Safari: 15.6
  npmPackages:
    @emotion/react: ^11.9.0 => 11.9.3 
    @emotion/styled: ^11.8.1 => 11.9.3 
    @mui/base:  5.0.0-alpha.91 
    @mui/icons-material: ^5.6.2 => 5.8.4 
    @mui/lab: ^5.0.0-alpha.79 => 5.0.0-alpha.92 
    @mui/material: ^5.6.3 => 5.9.2 
    @mui/private-theming:  5.9.1 
    @mui/styled-engine:  5.8.7 
    @mui/system:  5.9.2 
    @mui/types:  7.1.5 
    @mui/utils:  5.9.1 
    @mui/x-data-grid:  5.14.0 
    @mui/x-data-grid-premium: ^5.14.0 => 5.14.0 
    @mui/x-data-grid-pro:  5.14.0 
    @mui/x-date-pickers: ^5.0.0-beta.0 => 5.0.0-beta.2 
    @mui/x-license-pro:  5.14.0 
    @types/react: ^18.0.7 => 18.0.15 
    react: ^18.0.0 => 18.2.0 
    react-dom: ^18.0.0 => 18.2.0 
    typescript: ^4.6.3 => 4.7.4 
</details>

**Comment by alexfauquette** [comment]
Yes, this is explained in the doc. I keep this issue to track progress on this 

> This feature is not yet compatible with sortingMode = "server" and filteringMode = "server"

**Comment by vbynda-gresham** [comment]
We are also Premium plan customer, do you have ETA on when this may be looked at?

**Comment by nielsbrakel** [comment]
Running into the same problem, Premium user to.

**Comment by rettoua** [comment]
@alexfauquette The doc is saying about it in the `Single grouping column` section, but it doesn't work for `Multiple grouping columns` either. There is plenty of time has passed since the initial post; 
are there any changes or workaround to make it work eventually? 
p.s. my orderid is 54867, if so.

**Comment by michelengelen** [comment]
Thanks for the report and for everyone's patience here. Mixing client-side row grouping with `sortingMode="server"` was never made compatible: the grid grouped rows it had already received, while sorting happened elsewhere, which is why children rendered detached from their groups. The supported path today is server-side row grouping through the Data Source: the grid requests each group's children from your server with the sort and filter parameters included, so ordering and grouping stay consistent. See https://mui.com/x/react-data-grid/server-side-data/row-grouping/ (Premium).

Closing as completed. If the Data Source variant misbehaves with server sorting on the latest version, please open a new issue.
